core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -1323,6 +1326,42 @@ public boolean usesLegacyTableLayouts(Session session, TableHandle table)
                 });
     }
 
+    @Override
+    public Optional<JoinApplicationResult<TableHandle>> applyJoin(
+            Session session,
+            JoinType joinType,
+            TableHandle left,
+            TableHandle right,
+            List<JoinCondition> joinConditions,
+            Map<String, ColumnHandle> leftAssignments,
+            Map<String, ColumnHandle> rightAssignments)
+    {
+        if (!right.getCatalogName().equals(left.getCatalogName())) {
```

Comment:
If user query referce to `catalog` per `caTALog` or `"caTALog"`, will we get a normalized name here?

Reply:
I think so. The catalog name is put into `TableHandle` from `CatalogMetadata`. Let me double check though.

Reply:
It is good.

Reply:
Would you still want to have case insensitive comparison here, to (maybe) guard us from (some) future code changes?

Reply:
If we consider future changes, we could be concerned about catalog names becoming not only lowercase. Let's maybe add a comment, but no need to change the code.